File: Domain/Others/News.cs

namespace Domain.Others
{
    public class News
    {
        public string Id { get; set; } 
        public string Title { get; set; }
        public string SubTitle { get; set; }
        public string Body { get; set; }
        public bool IsHidden { get; set; } = false;
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public ICollection<NewsPhoto> NewsPhotos { get; set; } = new List<NewsPhoto>();
        public AppUser Author { get; set; }
    }
}

File: client-app/src/models/News.ts
import { RcFile } from "antd/es/upload";
import { Profile } from "./User";

export interface News {
  id: string;
  title: string;
  subTitle?: string;
  isHidden: boolean;
  createdAt: Date;
  mainImage: string;
}

export class NewsDetail {
  id;
  title;
  subTitle;
  isHidden;
  createdAt;
  mainImage;

  body?: string | undefined;
  newsPhotos?: NewsPhoto[] | undefined;
  author?: Profile | undefined;

  constructor(news: News) {
    this.id = news.id;
    this.title = news.title;
    this.subTitle = news.subTitle;
    this.isHidden = news.isHidden;
    this.createdAt = news.createdAt;
    this.mainImage = news.mainImage;
  }
}

export interface NewsPhoto {
  id: string;
  url: string;
  isMain: boolean;
  createdAt: Date;

  previewURL: string;
}

export class NewsCreate {
  title: string = "";
  subTitle?: string | null;
  body: string = "";
  fileImages: RcFile[] | undefined;

  newsPhotos?: NewsPhoto[];
}

export class NewsUpdate extends NewsCreate {
  id: string | null = null;
}
